plugins/ltac2/tac2core.ml

```diff
@@ -1043,10 +1043,44 @@ let () =
   define "time" (option string @-> closure @-> tac valexpr) @@ fun s f ->
   Proofview.tclTIME s (thaw f)
 
+let () =
+  define "timeout" (int @-> closure @-> tac valexpr) @@ fun i f ->
+    Proofview.tclTIMEOUT i (thaw f)
+
+let () =
+  define "timeoutf" (float @-> closure @-> tac valexpr) @@ fun f64 f ->
+    Proofview.tclTIMEOUTF (Float64.to_float f64) (thaw f)
+
 let () =
   define "check_interrupt" (unit @-> tac unit) @@ fun _ ->
   Proofview.tclCHECKINTERRUPT
 
+let () =
+  define "throw_bt" (exn @-> exninfo @-> tac valexpr) @@ fun (e,_) info ->
+    Proofview.tclLIFT (Proofview.NonLogical.raise (e, info))
+
+let () =
+  define "zero_bt" (exn @-> exninfo @-> tac valexpr) @@ fun (e,info0) info ->
+    let info =
+      match Exninfo.get info Tac2bt.backtrace with
+      | None -> info0
+      | Some bt -> Exninfo.add info0 Tac2bt.backtrace bt
+    in
```

Comment:
Why are we using info0 at all here?

Reply:
I'm not sure, @Janno wrote these functions initially, so he might know.

Reply:
I am honestly not sure. I suspect I blindly iterated on the code until it did what I thought it should do. Is the code wrong or just weird?

Reply:
It feels wrong that we are discarding most of the explicitly provided `info`
If there is no specific reason to use `info0` I would stop using it.

Reply:
Makes sense. Fixed it to just use `info`.